compiler/codegen/cg-expr.cpp

```diff
@@ -5279,6 +5279,38 @@ DEFINE_PRIM(STACK_ALLOCATE_CLASS) {
     ret = codegenCast(at, codegenAddrOf(tmp));
 }
 
+static
+void codegenCallMemset(GenRet dest, Type* type) {
+  GenInfo *info = gGenInfo;
+
+  GenRet size = codegenSizeof(type);
+
+  // Must call with real pointer arguments (not lvalue)
+  INT_ASSERT(dest.isLVPtr == GEN_VAL);
+  // And also above call should generate a value
+  INT_ASSERT(size.isLVPtr == GEN_VAL);
+
+  if (info->cfile) {
+    GenRet zero = codegenZero32();
+    codegenCall("memset", dest, zero, size);
+  } else {
+#ifdef HAVE_LLVM
+    llvm::ConstantInt* zero = info->irBuilder->getIntN(8, 0);
```

Comment:
The zeroes that we have seem to have different widths (8-bit here vs 32-bit for C). I don't think this should be an issue, but wanted to bring up in case it was inadvertent, or you think it may be an issue.

Reply:
I think it's correct. C's `memset` is odd in that it accepts a byte to fill with as an `int`. The LLVM intrinsic accepts a byte (`i8`).